### ISSUE TITLE
Send email to user's email address when a change is made via TO API

### ISSUE THREAD

**Issue by mitchell852** [apache/trafficcontrol#7457]
## This Feature Request affects these Traffic Control components:
- Traffic Ops
- Traffic Portal

## Description
For security reasons, it would be helpful to receive an email when a change is made by my user to notify me of any potential unauthorized changes. This email would only be sent on PUT/POST/DELETE api calls performed by my user. Not GETs. The email would be sent only if an email address is defined for the user.

In addition, the user should be able to opt out of emails but by default emails should be sent.

The contents of the email can simply be the change log content. ie.:

`
USER: username1234
SERVER: cdn-ec-02.cdn.foo.bar, ID: 1234, ACTION: updated

If this action was not performed by you, please contact your security representative
`

**Comment by ocket8888** [comment]
That seems extremely annoying. Imagine that I filter the servers table by those with a given Profile, then queue updates on those servers. I now have potentially hundreds of emails being sent to me. It's hard to imagine anyone not opting out of that.

**Comment by mitchell852** [comment]
> That seems extremely annoying. Imagine that I filter the servers table by those with a given Profile, then queue updates on those servers. I now have potentially hundreds of emails being sent to me. It's hard to imagine anyone not opting out of that.

yeah i see your point. bulk actions would result in bulk emails although bulk actions is not supported in TPv1 but i suppose they are in TO api if you build a script to do so.
